### PR TITLE
feat(buttons): add tokens and update shim

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
In `@cds/core` themes default versions of `group buttons` overlap their borders. This change will be fixed after token migration, because otherwise breaks standard NG theme.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by valentin-mladenov** [comment]
depends on #812

File: projects/angular/src/layout/nav/_header.clarity.scss [hunk 1]
```diff
@@ -155,6 +155,7 @@
       align-items: center;
       height: header-variables.$clr-header-height;
       text-decoration: none;
+      gap: 0;
```
Comment (williamernest) [inline projects/angular/src/layout/nav/_header.clarity.scss:158]: Is this (and the other gap values in vertical-nav) necessary? I'm not seeing a change when manipulating them in the preview

File: projects/angular/src/button/_buttons.clarity.scss [hunk 2]
```diff
@@ -122,6 +121,11 @@
     } @else {
       $fallbackColor: getSassButtonColor($button-type, box-shadow-color);
 
+      // ATM (after token migration)
```
Comment (williamernest) [inline projects/angular/src/button/_buttons.clarity.scss:124]: These comments are a great idea to keep track of changes we need to make later.

File: projects/angular/src/layout/nav/_header.clarity.scss (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/layout/nav/_header.clarity.scss:158]: They are necessary in core themes light, dark and high contrast.

The preview is not changing because uses hamburger icon which is not part of the link. The link icon is actually hidden you have to set it to visible.

The new implementation is introducing `gap` as maintaining spacing (8px) between parts of a button icons and text. The header and in other places are a special case that icons and text should not have gap spacing between them. The spacing is handled in different way. Adding `gap: 0` allow the component to use button in their special way.

Comment (valentin-mladenov) [reply projects/angular/src/layout/nav/_header.clarity.scss:158]: In header specifically the gap contradicts with specifically this `margin-right`: https://github.com/vmware-clarity/ng-clarity/blob/main/projects/angular/src/layout/nav/_header.clarity.scss#L170

File: tests/snapshots/button/button-group--variants-core-dark.png
Comment (williamernest) [inline tests/snapshots/button/button-group--variants-core-dark.png:1]: Can you update the PR description to call out this change to button groups? This shift may be a bit jarring for some users, so we should explain why this is changing and how to get the old style. 

Outside of that, this PR LGTM

Comment (valentin-mladenov) [reply tests/snapshots/button/button-group--variants-core-dark.png:1]: Description updated.

**Review by williamernest — APPROVED** [review]

File: projects/angular/src/button/_buttons.clarity.scss [hunk 3]
```diff
@@ -122,6 +121,11 @@
...
+      // Replace box-shadow section with commented section. Renaming all of clr-btn 'box-shadow' to 'click' color properties is needed.
+      // $buttonColor: 'clr-btn-' + $button-type + '-click-color';
+      // @include mixins.css-var(background-color, $buttonColor, $fallbackColor, variables.$clr-use-custom-properties);
+
       // IE/OldEdge
```
Comment (Jinnie) [inline projects/angular/src/button/_buttons.clarity.scss:129]: ATM we may want to remove all these, if they are only IE compatibility related.

Comment (valentin-mladenov) [reply projects/angular/src/button/_buttons.clarity.scss:129]: I'll add those to the JIRA tiket.

File: projects/angular/src/data/datagrid/_datagrid.clarity.scss [hunk 4]
```diff
@@ -1470,9 +1470,19 @@
...
+          clr-datagrid-column-toggle-gap,
+          variables.$clr_baselineRem_4px,
+          variables.$clr-use-custom-properties
+        );
+        @include mixins.css-var(
+          padding,
+          clr-datagrid-column-toggle-padding,
+          0 variables.$clr_baselineRem_6px,
+          variables.$clr-use-custom-properties
+        );
```
Comment (dtsanevmw) [inline projects/angular/src/data/datagrid/_datagrid.clarity.scss:1485]: Do we need both of these? It seems we didn't have the gap before and keeping either one of them should be good.

File: projects/angular/src/data/datagrid/_datagrid.clarity.scss [hunk 5]
```diff
@@ -1511,6 +1568,15 @@
             datagrid-variables.$clr-datagrid-column-toggle-text-hover-color,
             variables.$clr-use-custom-properties
           );
+          cds-icon,
+          clr-icon {
+            @include mixins.css-var(
+              color,
+              clr-datagrid-column-toggle-text-hover-color,
```
Comment (dtsanevmw) [inline projects/angular/src/data/datagrid/_datagrid.clarity.scss:1575]: Same here. Isn't `toggle-icon-color` better for this case?

File: projects/angular/src/data/datagrid/_datagrid.clarity.scss [hunk 6]
```diff
@@ -1533,6 +1599,15 @@
             datagrid-variables.$clr-datagrid-column-toggle-text-active-color,
             variables.$clr-use-custom-properties
           );
+          cds-icon,
+          clr-icon {
+            @include mixins.css-var(
+              color,
+              clr-datagrid-column-toggle-text-active-color,
```
Comment (dtsanevmw) [inline projects/angular/src/data/datagrid/_datagrid.clarity.scss:1606]: Same here. Isn't `toggle-icon-active-color` better for this case?

File: projects/angular/src/data/datagrid/_datagrid.clarity.scss (hunk 4 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/datagrid/_datagrid.clarity.scss:1485]: We need both for this iteration.

Tokens require `gap`. Original NG require `padding`.
Once we migrate the tokens only the gap will remain.

File: projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss [hunk 7]
```diff
@@ -77,6 +77,7 @@
     //Display
     display: flex;
     flex-direction: column;
+    gap: 0;
```
Comment (dtsanevmw) [inline projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss:80]: As discussed offline, these `gap: 0` probably wouldn't be needed when we revert the change for the anchor.

File: projects/angular/src/layout/nav/_header.clarity.scss (hunk 1 above)
Comment (dtsanevmw) [inline projects/angular/src/layout/nav/_header.clarity.scss:158]: As discussed offline, these `gap: 0` probably wouldn't be needed when we revert the change for the anchor.

File: projects/angular/src/layout/_login.clarity.scss [hunk 8]
```diff
@@ -238,6 +238,7 @@
           margin-top: variables.$clr_baselineRem_0_5;
           font-size: variables.$clr_baselineRem_0_583;
           text-align: center;
+          place-content: center;
```
Comment (dtsanevmw) [inline projects/angular/src/layout/_login.clarity.scss:241]: As discussed offline, this probably wouldn't be needed when we revert the change for the anchor.

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 9]
```diff
@@ -186,7 +197,14 @@
...
   .alert-text {
-    display: inline-block; //needed for IE11
+    @include mixins.css-var(
+      display,
+      clr-link-display,
+      inline-block,
+      //needed for IE11
+      variables.$clr-use-custom-properties
+    );
+
```
Comment (dtsanevmw) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:207]: Do we need that and how it's connected with the `link` or the buttons?

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 10]
```diff
@@ -247,6 +265,13 @@
     }
 
     .alert-actions {
+      @include mixins.css-var(
+        display,
+        clr-link-display,
+        inline-block,
+        //needed for IE11
+        variables.$clr-use-custom-properties
+      );
```
Comment (dtsanevmw) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:274]: Do we need that and how it's connected with the `link` or the buttons?

File: projects/angular/src/utils/_reboot.clarity.scss [hunk 11]
```diff
@@ -426,8 +426,19 @@
...
+    @include mixins.css-var(gap, clr-link-gap, variables.$clr_baselineRem_4px, variables.$clr-use-custom-properties);
+
+    &:link {
+      @include mixins.css-var(
+        color,
+        clr-link-color,
+        nav-variables.$clr-link-color,
+        variables.$clr-use-custom-properties
+      );
+    }
```
Comment (dtsanevmw) [inline projects/angular/src/utils/_reboot.clarity.scss:441]: Discussed offline to rever these

File: projects/angular/src/wizard/_wizard.clarity.scss [hunk 12]
```diff
@@ -332,6 +332,8 @@
     .clr-wizard-stepnav-link {
       width: 100%;
       display: flex;
+      place-content: normal;
+      gap: 0;
```
Comment (dtsanevmw) [inline projects/angular/src/wizard/_wizard.clarity.scss:336]: Probably won't needs these with the link change reverted.

**Review by dtsanevmw — CHANGES_REQUESTED** [review]
Left some comments and some of the details we discussed offline.

File: projects/angular/src/wizard/_wizard.clarity.scss (hunk 12 above)
Comment (valentin-mladenov) [reply projects/angular/src/wizard/_wizard.clarity.scss]: corrected in commit bf22d26

File: projects/angular/src/utils/_reboot.clarity.scss (hunk 11 above)
Comment (valentin-mladenov) [reply projects/angular/src/utils/_reboot.clarity.scss]: corrected in commit bf22d26

File: projects/angular/src/layout/_login.clarity.scss (hunk 8 above)
Comment (valentin-mladenov) [reply projects/angular/src/layout/_login.clarity.scss]: corrected in commit bf22d26

File: projects/angular/src/data/datagrid/_datagrid.clarity.scss [hunk 13]
```diff
@@ -1491,6 +1501,53 @@
           datagrid-variables.$clr-datagrid-column-toggle-text-color,
           variables.$clr-use-custom-properties
         );
+        cds-icon,
+        clr-icon {
+          @include mixins.css-var(
+            color,
+            clr-datagrid-column-toggle-text-color,
```
[the remark this answers is not in the document]
Comment (valentin-mladenov) [reply projects/angular/src/data/datagrid/_datagrid.clarity.scss]: corrected in commit bf22d26

File: projects/angular/src/data/datagrid/_datagrid.clarity.scss (hunk 5 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/datagrid/_datagrid.clarity.scss]: corrected in commit bf22d26

File: projects/angular/src/data/datagrid/_datagrid.clarity.scss (hunk 6 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/datagrid/_datagrid.clarity.scss]: corrected in commit bf22d26

File: projects/angular/src/data/datagrid/_datagrid.clarity.scss [hunk 14]
```diff
@@ -1491,6 +1501,53 @@
...
+          @include mixins.css-var(
+            color,
+            clr-btn-default-disabled-color,
+            variables.$clr-color-neutral-700,
+            variables.$clr-use-custom-properties
+          );
+          @include mixins.css-var(
+            background-color,
+            clr-datagrid-column-toggle-fill-color,
+            datagrid-variables.$clr-datagrid-column-toggle-fill-color,
```
[the remark this answers is not in the document]
Comment (valentin-mladenov) [reply projects/angular/src/data/datagrid/_datagrid.clarity.scss]: corrected in commit bf22d26

File: projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss (hunk 7 above)
Comment (valentin-mladenov) [reply projects/angular/src/layout/vertical-nav/_vertical-nav.clarity.scss]: corrected in commit bf22d26

File: projects/angular/src/layout/nav/_header.clarity.scss (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/layout/nav/_header.clarity.scss]: corrected in commit bf22d26

File: projects/angular/src/emphasis/alert/_alert.clarity.scss (hunk 10 above)
Comment (valentin-mladenov) [reply projects/angular/src/emphasis/alert/_alert.clarity.scss]: corrected in commit eb48c7f

File: projects/angular/src/emphasis/alert/_alert.clarity.scss (hunk 9 above)
Comment (valentin-mladenov) [reply projects/angular/src/emphasis/alert/_alert.clarity.scss]: corrected in commit eb48c7f

**Comment by valentin-mladenov** [comment]
Resolves #712
